### PR TITLE
Upgrade to build v0.10

### PR DESCRIPTION
**Pull request by natebosch**

### CODE REVIEW

**Comment by kevmoo** [comment]
Check your base – you're not on latest master

**Comment by natebosch** [comment]
merged latest

File: pubspec.yaml [hunk 1]
```diff
@@ -20,3 +20,13 @@ dev_dependencies:
   test: ^0.12.3
   _test_annotations:
     path: ./_test_annotations
+
+dependency_overrides:
+  build:
+    git:
+      url: git@github.com:dart-lang/build.git
```
Comment (kevmoo) [inline pubspec.yaml]: https url only, yo – @natebosch

Comment (natebosch) [inline pubspec.yaml]: 1 - why?
2 - It's not gonna survive the PR - I'll have build published before we merge this.

Comment (natebosch) [inline pubspec.yaml]: Ah - cause travis complains

File: test/constants_test.dart [hunk 2]
```diff
@@ -57,8 +57,7 @@ void main() {
           const Super() : super(aString: 'Super Hello');
         }
       ''');
-      constants = resolver
-          .getLibraryByName('test_lib')
+      constants = (await resolver.findLibraryByName('test_lib'))
```
Comment (jakemac53) [inline test/constants_test.dart]: do we not need to do the same teardown logic here?

File: test/span_for_element_test.dart [hunk 3]
```diff
@@ -20,7 +20,7 @@ void main() {
     ''',
       inputId: new AssetId('test_lib', 'lib/test_lib.dart'),
     );
-    library = resolver.getLibraryByName('test_lib');
+    library = await resolver.findLibraryByName('test_lib');
```
Comment (jakemac53) [inline test/span_for_element_test.dart]: Or here?

File: test/constants_test.dart (hunk 2 above)
Comment (natebosch) [inline test/constants_test.dart]: probably - I missed some of these if they didn't cause the tests to fail. Perhaps another argument for a breaking change in the signature in build_test.

Comment (natebosch) [inline test/constants_test.dart]: Done

File: test/span_for_element_test.dart (hunk 3 above)
Comment (natebosch) [inline test/span_for_element_test.dart]: Done

**Pull request merged**